onnxruntime/contrib_ops/cuda/custom_layer_norm.cu

```diff
@@ -0,0 +1,305 @@
+    int64_t n2,
+    U* invvars,
+    U* means);
+
+template <>
+void launch_custom_layer_norm<float, float>(
+    float* vals,
+    const float* residual,
+    const float* gamma,
+    const float* beta,
+    float epsilon,
+    int64_t n1,
+    int64_t n2,
+    float* invvars,
+    float* means) {
+  constexpr int threads = 256;
+
+  dim3 grid_dim(n1);
+
+  dim3 block_dim(threads);
+
+  // There are some limitations to call below functions, now just enumerate the situations.
+  if (n2 == 768)
+    custom_layer_norm<768, 3><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 512)
+    custom_layer_norm<512, 2><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 1024)
+    custom_layer_norm<1024, 4><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 1536)
+    custom_layer_norm<1536, 6><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 2048)
+    custom_layer_norm<2048, 8><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 2560)
+    custom_layer_norm<2560, 10><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else {
+    //shouldn't reach here
+    ORT_THROW("Unsupported hidden_dim for custom LayerNorm implementation.");
+  }
+}
+
+template <>
+void launch_custom_layer_norm<__half, float>(
+    __half* vals,
+    const __half* residual,
+    const __half* gamma,
+    const __half* beta,
+    float epsilon,
+    int64_t n1,
+    int64_t n2,
+    float* invvars,
+    float* means) {
+  constexpr int threads = 128;
+
+  dim3 grid_dim(n1);
+  dim3 block_dim(threads);
+
+  // There are some limitations to call below functions, now just enumerate the situations.
+  if (n2 == 768)
+    custom_layer_norm<384, 3><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 512)
+    custom_layer_norm<256, 2><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 1024)
+    custom_layer_norm<512, 4><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 1536)
+    custom_layer_norm<768, 6><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 2048)
+    custom_layer_norm<1024, 8><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else if (n2 == 2560)
+    custom_layer_norm<1280, 10><<<grid_dim, block_dim, 0, 0>>>(
+        vals, residual, gamma, beta, epsilon, invvars, means);
+  else {
+    //shouldn't reach here
+    ORT_THROW("Unsupported hidden_dim for custom LayerNorm implementation.");
+  }
+}
+
+}  //namespace cuda
```

Comment:
nit: looks like this file escaped the formatter, as well as onnxruntime/contrib_ops/cuda/custom_layer_norm.cuh

Reply:
i.e. no trailing newline

Reply:
using https://marketplace.visualstudio.com/items?itemName=xaver.clang-format, this wasn't caught